### PR TITLE
Deploy token contract

### PR DESCRIPTION
**Pull request by paulbellamy**
Fixes https://github.com/stellar/soroban-cli/issues/81

Adds 2 new subcommands:
- `$ soroban-cli token create --name "Example Token" --symbol "EXT" --decimal 7 --admin "G...5"`
  Deploys a token smart contract for a new soroban token.
 
- `$ soroban-cli token wrap --asset "USDC:G...5"`
  Deploys the smart contract wrapper for an existing Stellar classic asset.

### TODO

- [x] Support deploying the token contracts to a remote soroban-rpc server
  - If we are doing the `soroban-cli sandbox/remote` distinction, these subcommands should probably be nested under that
- [x] fix type conversion error with latest rs-soroban-env and rs-soroban-sdk
- [ ] Support actually invoking the deployed contracts in the sandbox
  - Moved to new issue: https://github.com/stellar/soroban-cli/issues/172
- [ ] Decide about renaming the subcommands
- [ ] Figure out how to test this
- [ ] update soroban-example-dapp: https://github.com/stellar/soroban-example-dapp/pull/20
  - ~One issue here, is that because #172 is not done yet,it doesn't actually work... :/~

### CODE REVIEW

File: src/soroban_rpc/mod.rs [hunk 1]
```diff
@@ -0,0 +1,135 @@
+use jsonrpsee_core::{client::ClientT, rpc_params};
```
Comment (leighmcculloch) [inline src/soroban_rpc/mod.rs:1]: I think you could name this mod simply `rpc`. No need to echo soroban, there's only one rpc server.

File: src/token/create.rs [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+        conflicts_with = "ledger-file",
+        requires = "private-strkey",
+        requires = "network-passphrase"
+    )]
+    rpc_server_url: Option<String>,
+    // TODO: we should probably use an env variable for security reasons
+    //       (otherwise it will get recorded in the shell history etc ...)
+    /// Private key to sign the transaction sent to the rpc server
+    #[clap(long = "private-strkey")]
+    private_strkey: Option<String>,
```
Comment (leighmcculloch) [inline src/token/create.rs:109]:
```suggestion
    /// Private key to sign the transaction sent to the rpc server
    #[clap(long = "private-strkey", env)]
    private_strkey: Option<String>,
```

File: src/token/wrap.rs [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+        conflicts_with = "ledger-file",
+        requires = "private-strkey",
+        requires = "network-passphrase"
+    )]
+    rpc_server_url: Option<String>,
+    // TODO: we should probably use an env variable for security reasons
+    //       (otherwise it will get recorded in the shell history etc ...)
+    /// Private key to sign the transaction sent to the rpc server
+    #[clap(long = "private-strkey")]
+    private_strkey: Option<String>,
```
Comment (leighmcculloch) [inline src/token/wrap.rs:89]:
```suggestion
    /// Private key to sign the transaction sent to the rpc server
    #[clap(long = "private-strkey", env)]
    private_strkey: Option<String>,
```

**Review by leighmcculloch — APPROVED** [review]

File: src/soroban_rpc/mod.rs (hunk 1 above)
Comment (paulbellamy) [reply src/soroban_rpc/mod.rs:1]: maybe rpc_client as well tbh

File: src/token/wrap.rs [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+        vec![ScVal::Object(Some(ScObject::Bytes(buf.try_into()?)))].try_into()?;
+
+    let op = Operation {
+        source_account: None,
+        body: OperationBody::InvokeHostFunction(InvokeHostFunctionOp {
+            function: HostFunction::CreateTokenContractWithAsset,
+            parameters: parameters.into(),
+            footprint: LedgerFootprint {
+                read_only: VecM::default(),
+                read_write: vec![lk].try_into()?,
```
Comment (sisuresh) [inline src/token/wrap.rs:213]: `init_wrap` will also write `metadata`. It will also write an `admin` if the token is not native.

**Review by 2opremio — APPROVED** [review]

**Pull request merged**